llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,392 @@
+
+    def __init__(self, *, client: Any, **kwargs: Any):
+        """Raw constructor.
+
+        Use the class method `from_corpus` or `create_corpus` instead.
+
+        Args:
+            client: The low-level retriever class from google.ai.generativelanguage.
+        """
+        try:
+            import google.ai.generativelanguage as genai
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        super().__init__(**kwargs)
+
+        assert isinstance(client, genai.RetrieverServiceClient)
+        self._client = client
+
+    @classmethod
+    def from_corpus(cls, *, corpus_id: str) -> "GoogleVectorStore":
+        """Create an instance that points to an existing corpus.
+
+        Args:
+            corpus_id: ID of an existing corpus on Google's server.
+
+        Returns:
+            An instance of the vector store that points to the specified corpus.
+
+        Raises:
+            NoSuchCorpusException if no such corpus is found.
+        """
+        try:
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(f"\n\nGoogleVectorStore.from_corpus(corpus_id={corpus_id})")
+        if genaix.get_corpus(corpus_id=corpus_id) is None:
+            raise NoSuchCorpusException(corpus_id=corpus_id)
+
+        return cls(corpus_id=corpus_id, client=genaix.build_semantic_retriever())
+
+    @classmethod
+    def create_corpus(
+        cls, *, corpus_id: Optional[str] = None, display_name: Optional[str] = None
+    ) -> "GoogleVectorStore":
+        """Create an instance that points to a newly created corpus.
+
+        Examples:
+            store = GoogleVectorStore.create_corpus()
+            print(f"Created corpus with ID: {store.corpus_id})
+
+            store = GoogleVectorStore.create_corpus(
+                display_name="My first corpus"
+            )
+
+            store = GoogleVectorStore.create_corpus(
+                corpus_id="my-corpus-1",
+                display_name="My first corpus"
+            )
+
+        Args:
+            corpus_id: ID of the new corpus to be created. If not provided,
+                Google server will provide one for you.
+            display_name: Title of the corpus. If not provided, Google server
+                will provide one for you.
+
+        Returns:
+            An instance of the vector store that points to the specified corpus.
+        """
+        try:
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(
+            f"\n\nGoogleVectorStore.create_corpus(new_corpus_id={corpus_id}, new_display_name={display_name})"
+        )
+
+        client = genaix.build_semantic_retriever()
+        new_corpus_id = corpus_id or str(uuid.uuid4())
+        new_corpus = genaix.create_corpus(
```

Comment:
This can fail if the corpus_id already exists or if the user has hit their quota.